File: backend/app/schemas/api_key.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class APIKeyBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)
    exchange: str = Field(..., description="交易所类型: okx/binance/...")
    api_key: str
    api_secret: str
    passphrase: Optional[str] = None
    is_demo: bool = True


class APIKeyCreate(APIKeyBase):
    pass


class APIKeyUpdate(BaseModel):
    name: Optional[str] = None
    api_key: Optional[str] = None
    api_secret: Optional[str] = None
    passphrase: Optional[str] = None
    is_demo: Optional[bool] = None
    is_active: Optional[bool] = None


class APIKeyResponse(BaseModel):
    id: int
    name: str
    exchange: str
    api_key_masked: str  # 脱敏后的API Key
    is_demo: bool
    is_active: bool
    last_verified_at: Optional[datetime] = None
    created_at: datetime

    class Config:
        from_attributes = True


class APIKeyVerifyResponse(BaseModel):
    success: bool
    message: str
    balance: Optional[float] = None

File: src/api/apiKey.ts
import apiClient from './request'

export interface APIKey {
  id: number
  name: string
  exchange: string
  api_key_masked: string
  is_demo: boolean
  is_active: boolean
  last_verified_at?: string
  created_at: string
}

export interface APIKeyCreate {
  name: string
  exchange: string
  api_key: string
  api_secret: string
  passphrase?: string
  is_demo: boolean
}

export interface APIKeyUpdate {
  name?: string
  api_key?: string
  api_secret?: string
  passphrase?: string
  is_demo?: boolean
  is_active?: boolean
}

export interface APIKeyVerifyResponse {
  success: boolean
  message: string
  balance?: number
}

// 获取API密钥列表
export function getAPIKeys(): Promise<APIKey[]> {
  return apiClient.get('/user/api-keys')
}

// 创建API密钥
export function createAPIKey(data: APIKeyCreate): Promise<APIKey> {
  return apiClient.post('/user/api-keys', data)
}

// 更新API密钥
export function updateAPIKey(keyId: number, data: APIKeyUpdate): Promise<APIKey> {
  return apiClient.put(`/user/api-keys/${keyId}`, data)
}

// 删除API密钥
export function deleteAPIKey(keyId: number): Promise<{ success: boolean; message: string }> {
  return apiClient.delete(`/user/api-keys/${keyId}`)
}

// 验证API密钥
export function verifyAPIKey(keyId: number): Promise<APIKeyVerifyResponse> {
  return apiClient.post(`/user/api-keys/${keyId}/verify`)
}
